libs/interpret-vision/src/lib/VisionExplanationDashboard/Controls/DataCharacteristics.tsx

```diff
@@ -227,8 +226,202 @@ export class DataCharacteristics extends React.Component<
     );
   };
 
+  private processData = (): void => {
+    const examples: IVisionListItem[] = this.props.data;
+    const columnCount: number[] = [];
+    const itemsP: Map<string, IVisionListItem[]> = new Map();
+    const itemsT: Map<string, IVisionListItem[]> = new Map();
+    const labelVisibilitiesP: Map<string, boolean> = new Map();
+    const labelVisibilitiesT: Map<string, boolean> = new Map();
+    const renderStartIndex: number[] = [];
+    const showBackArrow: boolean[] = [];
+    const dropdownOptionsP: IDropdownOption[] = [];
+    const dropdownOptionsT: IDropdownOption[] = [];
+    const selectedKeysP: string[] = [];
+    const selectedKeysT: string[] = [];
+    dropdownOptionsP.push({
+      key: SelectAllKey,
+      text: localization.InterpretVision.Dashboard.selectAll
+    });
+    dropdownOptionsT.push({
+      key: SelectAllKey,
+      text: localization.InterpretVision.Dashboard.selectAll
+    });
+    examples.forEach((example) => {
+      const labelP = example[labelTypes.predictedY].toString();
```

Comment:
nit: I would just refactor this code to loop over predicted/true y, since most of the code is the same.  That should cut down the lines of code by half I think.

Reply:
I would refactor it to a method and call it twice here